tests/core_tests/test_api_data/example_storage_root_hook_2/default_storage_root.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2023 Autodesk.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that gets executed during Toolkit initialization.
+This hook makes it possible to modify the default storage root.
```

Comment:
There's no difference since the hooks achieve the same thing. The difference is in the docstring of the execute function which specifies exactly how the root is retrieved.